// Copyright 2022-2024 ETSI SDG TeraFlowSDN (TFS) (https://tfs.etsi.org/)
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package optical_attack_detector;

import "context.proto";
import "monitoring.proto"; // to be migrated to: "kpi_manager.proto"

service OpticalAttackDetectorService {
  
  // rpc that executes the detection loop for a particular service
  rpc DetectAttack (DetectionRequest) returns (context.Empty) {}

}

message DetectionRequest {
  context.ServiceId service_id = 1;
  monitoring.KpiId  kpi_id     = 2; // to be migrated to: "kpi_manager.KpiId"
}
